lib/sentry/integrations/telemetry.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Sentry.Integrations.Telemetry do
+  @moduledoc """
+  Sentry integration for Telemetry.
+
+  *Available since v10.10.0*.
+  """
+
+  @moduledoc since: "10.10.0"
+
+  @failure_event [:telemetry, :handler, :failure]
+
+  @doc false
+  @spec attach() :: :ok
+  def attach do
+    _ =
+      :telemetry.attach(
+        "#{inspect(__MODULE__)}-telemetry-failures",
+        @failure_event,
+        &handle_event/4,
+        :no_config
+      )
+
+    :ok
+  end
+
+  @doc false
+  @spec handle_event(
+          :telemetry.event_name(),
+          :telemetry.event_measurements(),
+          :telemetry.event_metadata(),
+          :telemetry.handler_config()
+        ) :: :ok
+  def handle_event(@failure_event, _measurements, %{} = metadata, :no_config) do
+    stacktrace = metadata[:stacktrace] || []
+
+    options = [
+      stacktrace: stacktrace,
+      tags: %{
+        telemetry_handler_id: stringified_handler_id(metadata[:handler_id]),
+        event_name: inspect(metadata[:event_name])
+      }
+    ]
+
+    _ =
+      case {metadata[:kind], metadata[:reason]} do
+        {:error, reason} ->
+          exception = Exception.normalize(:error, reason, stacktrace)
+          Sentry.capture_exception(exception, options)
+
+        {kind, reason} ->
+          options =
+            Keyword.merge(options,
+              extra: %{kind: inspect(kind), reason: inspect(reason)},
+              interpolation_parameters: [inspect(metadata[:handler_id])]
+            )
+
+          Sentry.capture_message("Telemetry handler %s failed", options)
```

Comment:
Shouldn't this be a warning or even an error?

Reply:
This is already an error that gets reported to Sentry. Not sure what you mean here?

Reply:
@whatyouhide oh I just assumed that `capture_message` doesn't send an error by default, and sends an info-level message instead. IIRC Sentry displays and manages these messages differently, especially when it comes to notifications.

Reply:
I think you're thinking about logging. These get displayed as errors.

Reply:
@whatyouhide messages in Sentry do have levels too, and our SDK supports `level` option in `capture_message`. I just checked and luckily the default level is `error`.

One thing I noticed is that the UI displays "(no error message)" which may be a little confusing for the users.

<img width="744" alt="image" src="https://github.com/user-attachments/assets/425f49a0-dd46-4caf-a912-7b5edde553e7" />